### DISCUSSION TITLE
Query invalidation not working on mutate success

### DISCUSSION THREAD

**Discussion by vervex** [TanStack/query#2239] [Q&A]
I am using mutate and and on successful mutate I am calling query invalidate.

this gets fired for sure but doesn’t invalidate the query. I saw someone else was also having same issue in discord channel

**Comment by TkDodo** [comment]
please create a codesandbox reproduction. Most common gotchas are:
- new QueryClient() created _in_ the app component, which will create a new cache when re-rendering
- Using the wrong query key
